### PR TITLE
chore: Fix warnings produced by shellcheck on bench.sh

### PR DESCRIPTION
**Pull request by eejbyfeldt**
## Which issue does this PR close?

N/A

## Rationale for this change

Many of the warnings produced by shellcheck are indications of bugs in the script that would break under certain circumstances.

## What changes are included in this PR?

The following warnings was previously produced:
[SC2006](https://www.shellcheck.net/wiki/SC2006)
[SC2035](https://www.shellcheck.net/wiki/SC2035)
[SC2045](https://www.shellcheck.net/wiki/SC2045)
[SC2086](https://www.shellcheck.net/wiki/SC2086)
[SC2140](https://www.shellcheck.net/wiki/SC2140)
[SC2221](https://www.shellcheck.net/wiki/SC2221) ([SC2222](https://www.shellcheck.net/wiki/SC2222))

## Are these changes tested?

Manually running the script.

## Are there any user-facing changes?

No.

### CODE REVIEW

**Review by edmondop — COMMENTED** [review]
I wonder if we want to add something here https://github.com/apache/datafusion/tree/main/dev (`lint_scripts.sh`) maybe? And maybe also a task in CI?

**Comment by eejbyfeldt** [comment]
> I wonder if we want to add something here https://github.com/apache/datafusion/tree/main/dev (`lint_scripts.sh`) maybe? And maybe also a task in CI?

Agreed. If people agree that this is a change in the desired direction. Then it makes sense to address the same issues in other bash scripts in this repo and add a CI check as a follow up.

File: benchmarks/bench.sh [hunk 1]
```diff
@@ -189,7 +189,7 @@ main() {
             echo "DATA_DIR: ${DATA_DIR}"
             echo "RESULTS_DIR: ${RESULTS_DIR}"
             echo "CARGO_COMMAND: ${CARGO_COMMAND}"
-            echo "PREFER_HASH_JOIN": ${PREFER_HASH_JOIN}
+            echo "PREFER_HASH_JOIN: ${PREFER_HASH_JOIN}"
```
Comment (comphead) [inline benchmarks/bench.sh:192]: 👍

**Review by comphead — APPROVED** [review]
Thanks @eejbyfeldt I feel this PR is great.

**Comment by comphead** [comment]
> > I wonder if we want to add something here https://github.com/apache/datafusion/tree/main/dev (`lint_scripts.sh`) maybe? And maybe also a task in CI?
> 
> Agreed. If people agree that this is a change in the desired direction. Then it makes sense to address the same issues in other bash scripts in this repo and add a CI check as a follow up.

This is a nice idea. Not sure about having another CI task though.

**Review by edmondop — APPROVED** [review]

**Pull request merged**